test/system/030-run.bats

```diff
@@ -1104,5 +1104,13 @@ EOF
     rm -rf $romount
 }
 
+@test "podman run --restart=always -- wait" {
+    # regression test for #18572 to make sure Podman waits less than 20 seconds
+    ctr=$(random_string)
+    run_podman run -d --restart=always --name=$ctr $IMAGE false
+    PODMAN_TIMEOUT=20 run_podman wait $ctr
```

Comment:
This seems sketchy: shouldn't it be a little higher, say 22-23, to account for general system and startup lags?

Reply:
If it takes over 20 seconds it would mean the fix here is not working and we run into the built in timeout in the code which is exactly what we do not want.